### ISSUE TITLE
Delegation / Redelegation

### ISSUE THREAD

**Issue by NodeGuy** [cosmos/voyager#1219]
Jordan, please link your design files here.

**Comment by jbibla** [comment]
https://zpl.io/bWL0krn - Staking
https://zpl.io/VxlvdWk - Restaking

**Comment by NodeGuy** [comment]
Please remove issues from the backlog when moving them into Scrumptious.

**Comment by NodeGuy** [comment]
Hold on cowboys, this is dependent on #1171 which is also part of this sprint.

**Comment by jbibla** [comment]
it doesn't need to be dependent on #1171 - we can add a button to the existing profile page and work on the new modal ...

**Comment by faboweb** [comment]
I agree with Jordan. I would see this as a replacement for PageBond. We can add the 2 buttons to the old PageValidator for now and replace the checkboxes (and shoppingCart logic) already.

**Comment by jbibla** [comment]
yes, let's please get rid of `shoppingCart`.

🛒 🛒 🛒

**Comment by NodeGuy** [comment]
OK

**Comment by NodeGuy** [comment]
After looking at the designs it seems this isn't about redelegation—it's about delegation and undelegation, right?  I'm going to rename the title.

**Comment by jbibla** [comment]
we have to account for (`delegation / staking / bonding`), (`undelegation / unstaking / unbonding`), (`redelegation / restaking / rebonding`).

**Comment by faboweb** [comment]
Follow up for redelegation: https://github.com/cosmos/voyager/issues/1293

**Comment by NodeGuy** [comment]
Our standard modals have a darker background for the header:

<img width="581" alt="screen shot 2018-09-06 at 3 37 52 pm" src="https://user-images.githubusercontent.com/2148537/45180976-0985e180-b1eb-11e8-9c6d-67e80f12df42.png">

There's no such distinction in the design.  Do you care or should I reimplement `TmModal` for this one?

**Comment by NodeGuy** [comment]
Split out #1294 and #1295.

**Comment by faboweb** [comment]
the design for staking is actually not the same modal we have already. It is more a pop-over on the bottom like when you write a new email in google inbox. maybe it should still have the same layout. @jbibla can you help out?

**Comment by NodeGuy** [comment]
Oh, OK.  I see another difference: With `TmModal` the rest of the window is darkened but that's not happening here.  I won't use it.

**Comment by fedekunze** [comment]
> close window when clicking outside of it

@NodeGuy do we need this point if we are already adding the "Close" and `X` buttons ?

**Comment by NodeGuy** [comment]
> @NodeGuy do we need this point if we are already adding the "Close" and X buttons ?

@faboweb requested it and I think it's a good idea.

> So that would basically turn the delegation into a redelegation, right ? I'm asking because eventually we'll end up splitting that POST endpoint to submit delegations (see cosmos/cosmos-sdk#2191). So will have add this logic here as well

I guess so but haven't seen a spec for any of this.  @faboweb is there a spec?

**Comment by faboweb** [comment]
> So that would basically turn the delegation into a redelegation, right ?

Yep. Maybe do delegation first and then do redelegation in another PR.

> I'm asking because eventually we'll end up splitting that POST endpoint to submit delegations

Let's not handle problems we do not yet have. When the split happens, the logic on our hand probably can be easily changed.

**Comment by NodeGuy** [comment]
Created new issue about warning dialog: https://github.com/cosmos/voyager/issues/1321
